.github/workflows/basic.yml

```diff
@@ -12,13 +12,18 @@ jobs:
         build_type: [Debug, Release]
         compiler: [{c: gcc, cxx: g++}]
         pool_tracking: ['-DUMF_ENABLE_POOL_TRACKING=ON', '-DUMF_ENABLE_POOL_TRACKING=OFF']
+        shared_library: ['-DUMF_BUILD_SHARED_LIBRARY=OFF']
         include:
           - os: 'ubuntu-20.04'
             build_type: Release
             compiler: {c: gcc-7, cxx: g++-7}
           - os: 'ubuntu-22.04'
             build_type: Release
-            compiler: {c: clang, cxx: clang++}
+            compiler: {c: clang, cxx: clang++}            
+          - os: 'ubuntu-22.04'
+            build_type: Release
+            compiler: {c: gcc, cxx: g++}
```

Comment:
`compiler` line is redundant, I think. The main matrix doesn't include  other compilers than gcc.

Reply:
It is not redundant. It is needed, because it is used here:
```
          -DCMAKE_C_COMPILER=${{matrix.compiler.c}}
          -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}}
```

Reply:
I'm not sure why, but this workflow doesn't work if I omit setting the compiler even if I use the "default" values

Reply:
I see now. Because you added a 'global' `shared_library` variable for multiple jobs and you change this value to a value not present in the matrix, you have to define all variables here as this job's configuration can't be added to existing jobs. See [here](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations )